### PR TITLE
Test remote buckets WIP

### PR DESCRIPTION
**Pull request by adfost**
Test for adding remote bucket

This will test:

`~/console/swagger-console.yml`

```sh
  /remote-buckets:
    get:
      summary: List Remote Buckets
      operationId: ListRemoteBuckets
      responses:
        200:
          description: A successful response.
          schema:
            $ref: "#/definitions/listRemoteBucketsResponse"
        default:
          description: Generic error response.
          schema:
            $ref: "#/definitions/error"
      tags:
        - Bucket
    post:
      summary: Add Remote Bucket
      operationId: AddRemoteBucket
      parameters:
        - name: body
          in: body
          required: true
          schema:
            $ref: "#/definitions/createRemoteBucket"
      responses:
        201:
          description: A successful response.
        default:
          description: Generic error response.
          schema:
            $ref: "#/definitions/error"
      tags:
        - Bucket
```

### CODE REVIEW

File: integration/remote_buckets_test.go [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+	tests := []struct {
+		name           string
+		args           args
+		expectedStatus int
+		expectedError  error
+	}{
+		{
+			name: "Create Remote Bucket - Valid",
+			args: args{
+				api:          "/remote-buckets",
```
Comment (dvaldivia) [inline integration/remote_buckets_test.go:54]: all these tests are testing the same API, please remove this argument

**Review by dvaldivia — CHANGES_REQUESTED** [review]

File: integration/remote_buckets_test.go [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+	"time"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func Test_AddRemoteBucketAPI(t *testing.T) {
+	assert := assert.New(t)
+
+	AddBucket("remotebucketsource", false, true, map[string]interface{}{}, map[string]interface{}{})
+	AddBucketWithPort("remotebucketdest", false, true, map[string]interface{}{}, map[string]interface{}{}, "9001")
```
Comment (dvaldivia) [inline integration/remote_buckets_test.go:35]: no need to duplicate the code, just add some options to the `AddBucket` helper method to pass the port

**Review by dvaldivia — CHANGES_REQUESTED** [review]

File: integration/user_api_bucket_test.go [hunk 3]
```diff
@@ -69,6 +69,38 @@ func AddBucket(name string, locking bool, versioning bool, quota map[string]inte
 	return response, err
 }
 
+func AddBucketWithPort(name string, locking bool, versioning bool, quota map[string]interface{}, retention map[string]interface{}, port string) (*http.Response, error) {
```
Comment (dvaldivia) [inline integration/user_api_bucket_test.go:72]: this is duplicate code, just update the existing AddBucket to accept a optional options which can specify the port, this was also my previous comment

**Review by dvaldivia — CHANGES_REQUESTED** [review]

**Review by cniackz — CHANGES_REQUESTED** [review]
Get the token if you are going to use a different port other than `9090` otherwise `401` is observed while testing:

Screen Shot 2022-05-12 at 4 00 14 PM

**Comment by cniackz** [comment]
Hello Adam, good morning. Sorry I did not answer yesterday. I was busy with IBM Cloud Catalog.
Yes, I can see both buckets are being created and I see same error when testing with mc:
$ mc admin bucket remote add myminio/source http://minio:minio123@localhost/target --service "replication" --insecure
mc: <ERROR> Unable to configure remote target. Remote service connection error - please check remote service credentials and target bucket (Remote service endpoint or target bucket not available: target 
	Access Denied.).
Couple of comments:
If you plan on having two servers, I recommend separating them for they to work in parallel, because both in same function only one will work:
	// start console server
	go func() {
		fmt.Println("start server 9090")
		srv, err := initConsoleServer("9090")
		if err != nil {
			log.Println(err)
			log.Println("init fail")
			return
		}
		srv.Serve()
	}()
	go func() {
		fmt.Println("start server 9091")
		srv2, err := initConsoleServer("9091")
		if err != nil {
			log.Println(err)
			log.Println("init fail")
			return
		}
		srv2.Serve()

	}()
2. Second comment is, we only have one container for these tests:
@(docker run -v /data1 -v /data2 -v /data3 -v /data4 --net=mynet123 -d --name minio --rm -p 9000:9000 -p 9001:9001 -e MINIO_KMS_SECRET_KEY=my-minio-key:OSMM+vkKUTCvQs9YL/CVMIMt43HFhkUpqJxTmGl6rYw= $(MINIO_VERSION) server /data{1...4} --console-address ':9001' && sleep 5)
So don't use 9001 as it is collision with the container, try different port, also, I don't think this can be consider as remote because we are using the same container, we should be using two containers for the remote test. Nevertheless, I think we can safely use https://play.min.io for that matter as shown in:
https://github.com/cniackz/minio/wiki/remote-bucket
That way you don't need to create more containers, etc, etc... All you need is two create two buckets with versioning in both sides, in our current container, you can even reuse one if created and in play side with versioning, a simple mc command prior running the test can do the trick for us. This could simplify our testing a lot. But it is up to you as long as it works :slightly_smiling_face: and don't break other tests.

**Pull request closed without merging**